arm/Microsoft.DataFactory/factories/deploy.bicep

```diff
@@ -198,3 +209,6 @@ output dataFactoryResourceId string = dataFactory.id
 
 @description('The name of the Resource Group with the Data factory.')
 output dataFactoryResourceGroup string = resourceGroup().name
+
+@description('The principal ID of the system assigned identity.')
+output principalId string = systemAssignedIdentity ? dataFactory.identity.principalId : ''
```

Comment:
should we call is 'systemAssignedPrincipalId'? I wonder if people may wonder what the ID is from